swarmrl/tasks/object_movement/rod_rotation.py

```diff
@@ -85,17 +85,25 @@ def _compute_angular_velocity(self, new_director: np.ndarray):
                 Angular velocity of the rod
         """
         # Compute the angular velocity
-        dot_product = np.clip(
-            np.dot(self._historic_rod_director, new_director), -1.0, 1.0
+
+        # angle between the two directors
+        angular_velocity = np.arctan2(
+            self._historic_rod_director[0] * new_director[1]
+            - self._historic_rod_director[1] * new_director[0],
+            self._historic_rod_director[0] * new_director[0]
+            + self._historic_rod_director[1] * new_director[1],
         )
-        angular_velocity = np.arccos(dot_product)
 
         # Update the historical rod director
         self._historic_rod_director = new_director
-        velocity_change = angular_velocity - self._historic_velocity
+        velocity_change = abs(angular_velocity - self._historic_velocity)
```

Comment:
why do you do the abs here and later multiply the sign again?

Reply:
The value of the delta will dictate the reward, so if it is negative, it will be a punishment. I need to keep the signs in the first part because I need to know if it was spinning clockwise and then counter-clockwise which would reduce the value. However, if it was spinning -ve and then went more negative it would be a negative number overall so I take the abs after. That is the first line.

Second line is not really necessary it is just saying if you happened to have sped up, but changed direction while doing so, that was dumb don't do it. But I realise this morning after some sleep that this part is essentially enforcing that only one direction of rotation is chosen so it should be a kwarg.